video/cloud-client/analyze/beta_snippets.py

```diff
@@ -0,0 +1,207 @@
+        positions = '{}s to {}s'.format(start_time, end_time)
+        print('\tSegment: {}\n'.format(positions))
+
+        # There are typically many frames for each face,
+        # here we print information on only the first frame.
+        frame = face.frames[0]
+
+        time_offset = (frame.time_offset.seconds +
+                       frame.time_offset.nanos / 1e9)
+        box = frame.attributes[0].normalized_bounding_box
+
+        print('First frame time offset: {}s\n'.format(time_offset))
+
+        print('First frame normalized bounding box:')
+        print('\tleft  : {}'.format(box.left))
+        print('\ttop   : {}'.format(box.top))
+        print('\tright : {}'.format(box.right))
+        print('\tbottom: {}'.format(box.bottom))
+        print('\n')
+# [END video_face_bounding_boxes]
+
+
+# [START video_face_emotions]
+def face_emotions(path):
+    """ Analyze faces' emotions over frames. """
+    video_client = videointelligence.VideoIntelligenceServiceClient()
+    features = [videointelligence.enums.Feature.FACE_DETECTION]
+
+    # include_bounding_boxes must be set to True for include_emotions
+    # to work.
+    config = videointelligence.types.FaceConfig(
+        include_bounding_boxes=True,
+        include_emotions=True)
+    context = videointelligence.types.VideoContext(
+        face_detection_config=config)
+
+    operation = video_client.annotate_video(
+        path, features=features, video_context=context)
+    print('\nProcessing video for face annotations:')
+
+    result = operation.result(timeout=600)
+    print('\nFinished processing.')
+
+    # There is only one result because a single video was processed.
+    faces = result.annotation_results[0].face_detection_annotations
+    for i, face in enumerate(faces):
+        print('Face {}'.format(i))
+
+        frame_emotions = []
+        for frame in face.frames:
+            time_offset = (frame.time_offset.seconds +
+                           frame.time_offset.nanos / 1e9)
+            emotions = frame.attributes[0].emotions
+
+            # from videointelligence.enums
+            emotion_labels = (
+                'EMOTION_UNSPECIFIED', 'AMUSEMENT', 'ANGER',
+                'CONCENTRATION', 'CONTENTMENT', 'DESIRE',
+                'DISAPPOINTMENT', 'DISGUST', 'ELATION',
+                'EMBARRASSMENT', 'INTEREST', 'PRIDE', 'SADNESS',
+                'SURPRISE')
+
+            emotion, score = sorted(
+                [(em.emotion, em.score) for em in emotions],
+                key=lambda p: p[1])[-1]
+            emotion_label = emotion_labels[emotion]
+
+            frame_emotions.append((time_offset, emotion_label, score))
+
+        frame_emotions = sorted(frame_emotions, key=lambda p: p[0])
+        for time_offset, emotion_label, score in frame_emotions:
+            print('\t{:04.2f}s: {:14}({:4.3f})'.format(
+                time_offset, emotion_label, score))
+        print('\n')
```

Comment:
Should we merely print out the results without sorting based on score? Will help simplify the code, if we were going across the whole segment to determine the most likely emotion across all frames I think that would be cool, but not as simple.

Also by printing out the `time_offset`, isn't it always the same, since we only get the one frame for each face?

Reply:
Here actually we are sorting by time_offset - an earlier version of of the API did not sort the frames by it.  Emotion is a frame level output since detecting the change of emotions is useful.

The output might look like this:
```
	30.56s: INTEREST      (0.606)
	30.60s: INTEREST      (0.582)
	30.63s: INTEREST      (0.561)
	30.66s: AMUSEMENT     (0.559)
	30.70s: AMUSEMENT     (0.556)
	30.73s: AMUSEMENT     (0.549)
	30.76s: CONTENTMENT   (0.564)
	30.80s: CONTENTMENT   (0.634)
	30.83s: CONTENTMENT   (0.688)
```